apiserver/authentication/jwt/jwt.go

```diff
@@ -79,7 +80,26 @@ func NewAuthenticatorWithHTTPClient(
 
 // Authenticate implements EntityAuthenticator
 func (j *JWTAuthenticator) Authenticate(req *http.Request) (authentication.AuthInfo, error) {
-	return authentication.AuthInfo{}, fmt.Errorf("jwt authenticate %w", errors.NotImplemented)
+	authHeader := req.Header.Get("Authorization")
+	if authHeader == "" {
+		return authentication.AuthInfo{}, errors.NotFoundf("authorization header missing")
+	}
+
+	parts := strings.Fields(authHeader)
+	if len(parts) != 2 || parts[0] != "Bearer" {
```

Comment:
This could panic if parts length is 0. Can we ensure that there is a guard around this?

Reply:
I could be wrong but if the length is not 2 the second condition won't be evaluated.

Reply:
Ah true.